### PR TITLE
llvm => 18.1.0

### PR DESCRIPTION
**Pull request by satmandu**
- Also rebuilds `flex` to fix regression preventing `mesa` build.
- `sommelier` updated and tested on `x86_64`... but not on an arm device.
- `mesa` also rebuilt to depend upon llvm18.

Builds/Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` (Someone needs to test sommelier on here.)

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=llvm18 crew update
```

### CODE REVIEW

**Comment by satmandu** [comment]
Unit tests will fail until the final llvm18 packages still being built get added to this PR!

File: bin/crew [hunk 1]
```diff
@@ -1904,7 +1904,7 @@ def build_command(args)
     # Process preflight block to see if package should be built
     pre_flight
 
-    if !@pkg.is_fake? && @pkg.compatible? && @pkg.source?(ARCH) && @pkg.source_url.to_s.upcase != 'SKIP' && !@pkg.no_compile_needed?
+    if !@pkg.is_fake? && @pkg.compatible? && @pkg.source?(ARCH) && ( @pkg.no_source_build? || @pkg.source_url.to_s.upcase != 'SKIP' ) && !@pkg.no_compile_needed?
```
Comment (uberhacker) [inline bin/crew:1907]: Add `no_source_build` property help to `commands/help.rb`?

Comment (satmandu) [reply bin/crew:1907]: I broke this out into a separate PR: https://github.com/chromebrew/chromebrew/pull/9444

File: packages/glslang.rb [hunk 2]
```diff
@@ -14,9 +14,9 @@ class Glslang < CMake
   binary_compression 'tar.zst'
 
   binary_sha256({
-    aarch64: '3e751f7dd531c74756289df26b246b60ccbc366acdfe858fed75f0110fed70df',
-     armv7l: '3e751f7dd531c74756289df26b246b60ccbc366acdfe858fed75f0110fed70df',
-     x86_64: 'ef0a720e22567f8d14b2ed5dc14b5c489c4b5f5d9327847519b8299bf063fb19'
+    aarch64: 'd56db244e1e6178c97608af789c1a30f649885225aae8062a02178caf30bb5a8',
+     armv7l: 'd56db244e1e6178c97608af789c1a30f649885225aae8062a02178caf30bb5a8',
+     x86_64: 'cdee878c3b18ed1091b08d24defc57fb08b2b420348d43248d0c3ced29651070'
```
Comment (uberhacker) [inline packages/glslang.rb:19]: Bump version?

Comment (satmandu) [reply packages/glslang.rb:19]: Bumping versions is no longer needed for packages unless you want to preserve old binaries.

**Review by uberhacker — APPROVED** [review]

**Pull request merged**